src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -553,6 +553,53 @@ impl Step for Rustdoc {
         }
         let target = target_compiler.host;
 
+        let bin_rustdoc = || {
+            let sysroot = builder.sysroot(target_compiler);
+            let bindir = sysroot.join("bin");
+            t!(fs::create_dir_all(&bindir));
+            let bin_rustdoc = bindir.join(exe("rustdoc", target_compiler.host));
+            let _ = fs::remove_file(&bin_rustdoc);
+            bin_rustdoc
+        };
+
+        // If CI rustc is enabled and we haven't modified the rustdoc sources,
+        // use the precompiled rustdoc from CI rustc's sysroot to speed up bootstrapping.
+        if builder.download_rustc()
+            && target_compiler.stage > 0
+            && builder.rust_info().is_managed_git_subrepository()
+        {
+            let commit = get_closest_merge_base_commit(
+                Some(&builder.config.src),
+                &builder.config.git_config(),
+                &builder.config.stage0_metadata.config.git_merge_commit_email,
+                &[],
+            )
+            .unwrap();
+
+            let librustdoc_src = builder.config.src.join("src/librustdoc");
+            let rustdoc_src = builder.config.src.join("src/tools/rustdoc");
+            let has_changes = !t!(git(Some(&builder.config.src))
+                .args(["diff-index", "--quiet", &commit])
+                .arg("--")
+                .arg(librustdoc_src)
+                .arg(rustdoc_src)
+                .command
+                .status())
+            .success();
```

Comment:
```suggestion
            let has_changes = !git(Some(&builder.config.src))
                .allow_failure()
                .args(["diff-index", "--quiet", &commit])
                .arg("--")
                .arg(librustdoc_src)
                .arg(rustdoc_src)
                .run(builder
                .is_success();
```
:innocent:

Reply:
[Updated](https://github.com/rust-lang/rust/compare/154574c957c4b46cb65629c1b015c5f1184ba68c..474ac560ca5d17c78afc13bb32d1587f2e7fdeb0).

I removed the `BootstrapCommand::allow_failure` part and also used `CommandOutput::status` for a better equivalent.

Reply:
Hmm, but without `allow_failure` there is no point in asking if it has succeeded, because it will crash bootstrap on failure.